- refactor test code after code review of User Authentication feature

poula22 opened the pull request:
- make authentication class has 2 behaviors only login and logout
- move dependency of cacheRepository from logic layer to data layer
- create MateDataSource for add or delete or edit mate
- delete SignupUseCase.kt as it is not used
- delete ClearLoggedInUserFromCacheUseCase.kt as it behave as LogoutUseCase
- refactor test cases
- add test cases for CredentialValidator

poula22 committed:
- refactor test code after code review of User Authentication feature

ZainabAlnajjar commented on src/test/kotlin/logic/useCases/ChangePasswordUseCaseTest.kt:
```diff
@@ -3,9 +3,11 @@ package logic.useCases
 import io.mockk.mockk
-import logic.exceptions.AuthenticationException
+import logic.exceptions.InvalidPasswordException
+import logic.exceptions.InvalidUserNameLengthException
```
length is not the only thing important about usernames, this should be  `InvalidUserNameException` and should check for no white space and no comma in username, also password should check for  not having these

ZainabAlnajjar replied:
lets leave this for another PR dedicated for exceptions

poula22 committed:
Merge remote-tracking branch 'refs/remotes/origin/dev' into refactor/user-authentication

poula22 committed:
- make authentication class has 2 behaviors only login and logout
- move dependency of cacheRepository from logic layer to data layer
- create MateDataSource for add or delete or edit mate
- delete SignupUseCase.kt as it is not used
- delete ClearLoggedInUserFromCacheUseCase.kt as it behave as LogoutUseCase
- refactor test cases
- add test cases for CredentialValidator

ZainabAlnajjar commented:
since there are so many conflicts and most of this PR was solved in other PRs, we will get back to this PR once we finish other stuff that are more important
